### PR TITLE
Add new commandlet, Get-PnPExpiringMicrosoft365Groups

### PR DESCRIPTION
**Pull request by koskila**
Before creating a pull request, make sure that you have read the contribution file located at

https://github.com/pnp/powerShell/blob/dev/CONTRIBUTING.md

## Type ##
- [ ] Bug Fix
- [X] New Feature
- [ ] Sample

## Related Issues? ##
None.

## What is in this Pull Request ? ##
This PR adds new commandlet, Get-PnPExpiringMicrosoft365Groups. By default, it'll return groups expiring in the next 31 days (to be in line with SharePoint/OneDrive's retention period's 31-day months), but this can be changed by supplying a value to -Limit parameter.

I had to pass the additionalheaders to a few places to enable "advanced query capabilities", but I tried to comment that in the code to make it clear why.

### Guidance ###
* You can delete this section when you are submitting the pull request.* 
* *Please update this PR information accordingly. We use this as part of our release notes in monthly communications.*
* **Please target your PR to Dev branch. If you do not target the Dev branch we will not accept this PR.**

### CODE REVIEW

**Comment by gautamdsheth** [comment]
Awesome stuff @koskila, really glad to have you contributing here 🙏😊

Couple of things:

1) Can you please use singular, i.e rename it to `Get-PnPMicrosoft365ExpiringGroup`? It is a PowerShell convention. I know it's a bit weird, but that's how it is :(

2) Can you also please add the documentation for this? You can create a `Get-PnPMicrosoft365ExpiringGroup.md` file inside the `documentation` folder. You can check other docs for example.

File: src/Commands/Utilities/REST/GraphHelper.cs [hunk 1]
```diff
@@ -92,21 +92,17 @@ public static async Task<HttpResponseMessage> GetResponseAsync(PnPConnection con
...
             {
                 results.AddRange(request.Items);
                 while (!string.IsNullOrEmpty(request.NextLink))
                 {
-                    request = await GetAsync<RestResultCollection<T>>(connection, request.NextLink, accessToken, camlCasePolicy, propertyNameCaseInsensitive);
-                    if (request.Items.Any())
-                    {
-                        results.AddRange(request.Items);
-                    }
+                    request = await GetAsync<RestResultCollection<T>>(connection, request.NextLink, accessToken, camlCasePolicy, propertyNameCaseInsensitive, additionalHeaders);
```
Comment (gautamdsheth) [inline src/Commands/Utilities/REST/GraphHelper.cs:105]: Any reason you removed this portion ?

```
if (request.Items.Any())
                    {
                        results.AddRange(request.Items);
                    }
```

Comment (koskila) [reply src/Commands/Utilities/REST/GraphHelper.cs:105]: Unintentional change left over from debugging - thanks for catching it :D I'll restore it.

Comment (koskila) [reply src/Commands/Utilities/REST/GraphHelper.cs:105]: Oversight on my part - returning it. 😅

**Comment by gautamdsheth** [comment]
Thank you @koskila, have merged it !!

**Pull request merged**